### PR TITLE
Add some questions about http/php

### PR DESCRIPTION
**Pull request by alxvgt**
Here some questions about http to complete this category. Some other complementary quetions may be submitted in near future ;-)

### CODE REVIEW

**Comment by alxvgt** [comment]
@20uf Is it possible to use some piece of code well formatted in questions ? If yes, could you give an example ?

**Comment by lex111** [comment]
@alxvgt If I understand correctly, you can do it as shown [here](https://github.com/certificationy/php-pack/blob/master/data/basics.yml#L40).

**Comment by alxvgt** [comment]
> @alxvgt If I understand correctly, you can do it as shown [here](https://github.com/certificationy/php-pack/blob/master/data/basics.yml#L40).

@lex111 Thank you for your example, it will be useful in future ;-)

File: data/http.yml [hunk 1]
```diff
@@ -98,3 +107,74 @@ questions:
             - {value: "error404.html.twig",    correct: true}
             - {value: "404.html.twig",   correct: false}
             - {value: "twig.404.html.twig",   correct: false}
+    -
+        question: 'Which of the following HTTP headers can be used by the server to the client about cookies ?'
```
[the remark this answers is not in the document]
Comment (alxvgt) [reply data/http.yml]: a fix proposal is pushed, sorry for my low level english

File: data/http.yml [hunk 2]
```diff
@@ -98,3 +107,74 @@ questions:
...
+        answers:
+            - {value: 'Cookie', correct: false}
+            - {value: 'Set-Cookie', correct: true}
+            - {value: 'Delete-Cookie', correct: false}
+        help: |
+            https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
+            https://developer.mozilla.org/fr/docs/Web/HTTP/Cookies
+    -
+        question: 'Which of the following HTTP headers can be used by the server to the client about content negocitation ?'
```
[the remark this answers is not in the document]
Comment (alxvgt) [reply data/http.yml]: oops, a fix is pushed ;-)

File: data/http.yml [hunk 3]
```diff
@@ -98,3 +107,74 @@ questions:
...
+            - {value: 'PUT', correct: true}
+            - {value: 'DELETE', correct: true}
+            - {value: 'OPTIONS', correct: true}
+            - {value: 'PATCH', correct: false}
+            - {value: 'TRACE', correct: true}
+        help: |
+            https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html
+            https://tools.ietf.org/html/rfc7231#section-4.2.2
+    -
+        question: 'Which of the following HTTP methods are idempotent ?'
```
[the remark this answers is not in the document]
Comment (alxvgt) [reply data/http.yml]: oops, a fix is pushed, thx !

File: data/http.yml [hunk 4]
```diff
@@ -71,6 +71,15 @@ questions:
             - {value: 302, correct: true}
             - {value: 201, correct: false}
             - {value: 204, correct: false}
+    -
+        question: 'Which HTTP status code should for a resource that moved permanently ?'
```
[the remark this answers is not in the document]
Comment (alxvgt) [reply data/http.yml]: a fix proposal is pushed, sorry for my low level english...

**Comment by lex111** [comment]
@alxvgt thank you!

**Pull request merged**